runtime/calamari/src/lib.rs

```diff
@@ -891,7 +887,7 @@ pub type Executive = frame_executive::Executive<
     Block,
     frame_system::ChainContext<Runtime>,
     Runtime,
-    AllPalletsReversedWithSystemFirst,
```

Comment:
Why not to stick to `AllPalletsReversedWithSystemFirst`?

Reply:
AllPalletsReversedWithSystemFirst is deprecated, I add some important changes related upstream PR on the description.

Reply:
This might actually require some reordering of the construct_runtime macro.
I have to refresh my memory but weren't some of the pallets dependent on others, so they have to be in specific order .
Especially the consensus related ones @Garandor

Reply:
``` 
        ParachainStaking: pallet_parachain_staking::{Pallet, Call, Storage, Event<T>, Config<T>} = 48,
        // Collator support.
        AuthorInherent: pallet_author_inherent::{Pallet, Call, Storage, Inherent} = 60,
        AuraAuthorFilter: pallet_aura_style_filter::{Pallet, Storage} = 63,
        // The order of the next 4 is important and shall not change.
        Authorship: pallet_authorship::{Pallet, Call, Storage} = 20,
        CollatorSelection: manta_collator_selection::{Pallet, Call, Storage, Event<T>, Config<T>} = 21,
        Session: pallet_session::{Pallet, Call, Storage, Event, Config<T>} = 22,
        Aura: pallet_aura::{Pallet, Storage, Config<T>} = 23,
```
        
One thing im noticing is that during pallet-authorship's on_initialize hook it calls manta_collator_selection::not_author(), which mutates some storage in manta_collator_selection. But since manta_collator_selection doesn't do anything in its own on_init hook this won't matter.

Reply:
The other thing i noticed is that pallet-author-inherent uses `<frame_system::Pallet<T>>::digest();` in on_init but system is still first, so should be good.

Reply:
All the rest non-consensus were not brought up the last time we looked into this, so they should be ok.

Anyway nothing is showing up in the CI, and we still have the testnet and manual QA of staking to maybe catch something.